fission/flake.nix

```
@@ -0,0 +1,42 @@
+{
+  description = "Synthesis' Web-Based Robot Simulator";
+
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+
+  outputs =
+    inputs:
+    let
+      supportedSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+      forEachSupportedSystem =
+        f:
+        inputs.nixpkgs.lib.genAttrs supportedSystems (
+          system:
+          f {
+            pkgs = import inputs.nixpkgs { inherit system; };
+          }
+        );
+    in
+    {
+      devShells = forEachSupportedSystem (
+        { pkgs }:
```

pkgs is all you need here, the only other thing you might wanna access is the current system but that can be done via `pkgs.system`
```suggestion
        inputs.nixpkgs.lib.genAttrs supportedSystems (
          system:
          f nixpkgs.legacyPackages.${system}
        );
    in
    {
      devShells = forEachSupportedSystem (pkgs:
```